### DISCUSSION TITLE
Can't boot ubuntu focal cloud image by cloud-hypervisor 0.12

### DISCUSSION THREAD

**Discussion by jacky-lau-1** [cloud-hypervisor/cloud-hypervisor#2222] [Q&A] [answered]
Hi, all.
I download cloud-hypervisor 0.12 binary from https://github.com/cloud-hypervisor/cloud-hypervisor/releases/tag/v0.12.0 
and following the guide on your project's homepage. But unfortunately I can't boot ubuntu focal cloud image.
I use this command to run cloud-hypervisor:
./cloud-hypervisor-static --kernel ./hypervisor-fw-0.2.9 --disk path=focal-server-cloudimg-amd64.raw --cpus boot=4 --memory size=1024M --net "tap=ich0,mac=a4:a1:c2:00:00:01,ip=192.168.4.2,mask=255.255.255.0,num_queues=2,queue_size=256" --rng  --serial tty --console null --log-file /tmp/vmlog.txt -vvv

cloud-hypervisor-static output messages:
Cloud Hypervisor Guest
        API server: /run/user/1000/cloud-hypervisor.1696
        vCPUs: 4
        Memory: 1024 MB
        Kernel: Some(KernelConfig { path: "./hypervisor-fw-0.2.9" })
        Initramfs: None
        Kernel cmdline: 
        Disk(s): Some([DiskConfig { path: Some("focal-server-cloudimg-amd64.raw"), readonly: false, direct: false, iommu: false, num_queues: 1, queue_size: 128, vhost_user: false, vhost_socket: None, poll_queue: true, id: None, disable_io_uring: false }])

Booting with PVH Boot Protocol
Setting up 4 GiB identity mapping
Page tables setup
Found PCI device vendor=8086 device=d57 in slot=0
Found PCI device vendor=1af4 device=1043 in slot=1
Found PCI device vendor=1af4 device=1042 in slot=2
Found PCI device vendor=1af4 device=1041 in slot=3
Found PCI device vendor=1af4 device=1044 in slot=4
PCI Device: 0:2.0 1af4:1042
Bar: type=MemorySpace32 address=e7f80000
Bar: type=MemorySpace32 address=0
Bar: type=MemorySpace32 address=0
Bar: type=MemorySpace32 address=0
Bar: type=MemorySpace32 address=0
Bar: type=Unused address=0
Virtio block device configured. Capacity: 4612096 sectors
Found EFI partition
Filesystem ready
Error loading default entry: FileError(NotFound)
Using EFI boot.
Found bootloader (BOOTX64.EFI)
Executable loaded
Failed to set MokListRT: Unsupported
Could not create MokListRT: Unsupported
Something has gone seriously wrong: import_mok_state() failed
: Unsupported
TPM logging failed: Unsupported
Failed to allocate memory for 'gdt'
efi_main() failed!

[vmlog.txt.gz](https://github.com/cloud-hypervisor/cloud-hypervisor/files/5904075/vmlog.txt.gz)

**Comment by rbradford** [comment, marked as answer]
Hi @jacky-lau-1. This is due to an issue in the upstream Ubuntu image. You can see https://github.com/cloud-hypervisor/rust-hypervisor-firmware/issues/77 for details.

I recommend you try using this alternative: https://cloudhypervisorstorage.blob.core.windows.net/images/focal-server-cloudimg-amd64-custom.qcow2 or https://cloudhypervisorstorage.blob.core.windows.net/images/focal-server-cloudimg-amd64-custom.raw

**Reply by jacky-lau-1** [reply]
OK, thanks

**Comment by rbradford** [comment]
Rust Hypervisor Firmware 0.3.0: https://github.com/cloud-hypervisor/rust-hypervisor-firmware/releases/download/0.3.0/hypervisor-fw now lets you use stock Ubuntu images.
